### PR TITLE
Clean up Install Agent Flow on Linux docs

### PR DESCRIPTION
**Pull request by clayton-cornell**
#### PR Description

This is a general rework of the Agent Flow installation instructions for Linux:

1. Add uninstall steps for all supported distros
2. Clean up Fedora/RHEL repos steps
3. Switch from YUM to DNF in Fedora/RHEL (dnf replaced yum in 2015)
4. Clean up wording to make consistent
5. Add missing GPG steps

#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: docs/sources/flow/install/linux.md [hunk 1]
```diff
@@ -9,35 +9,64 @@ You can install Grafana Agent Flow as a systemd service on Linux.
...
+   sudo systemctl stop grafana-agent-flow
+   ```
+
+1. Uninstall Grafana Agent Flow:
 
+   ```shell
+   sudo apt remove grafana-agent-flow
    ```
+
+1. Optional. Remove the Grafana repository:
```
Comment (rfratto) [inline docs/sources/flow/install/linux.md:50]: The recommended syntax for [optionals](https://developers.google.com/style/procedures#optional-steps) is to use a colon following the optional indicator: 

```suggestion
1. Optional: Remove the Grafana repository:
```

Ditto in all the other places optional is used in this change

Comment (rfratto) [reply docs/sources/flow/install/linux.md:50]: Should we note that this should only be done if no other packages are installed from that repository?

**Review by rfratto — APPROVED** [review]
LGTM with one small piece of feedback

File: docs/sources/flow/install/linux.md (hunk 1 above)
Comment (osg-grafana) [reply docs/sources/flow/install/linux.md:50]: Stylistically, I have use `(Optional) Remove the Grafana repository` with no colon. This is something I can bring to the style meeting with the Docs Team. (Please use my version for now.)

Comment (clayton-cornell) [reply docs/sources/flow/install/linux.md]: Good point. I hadn't got around to the final style pass :-) A quick grep of the Grafana/Grafana and Agent repos... and `(Optional)` is the one we are using everywhere vs. colons.

**Review by chri2547 — APPROVED** [review]

**Comment by clayton-cornell** [comment]
Backport to v0.34 fixed
Cancelling backport to v0.32. Too many changes in the content to make it worth the effort to clean up.

**Pull request merged**